fabtests/common/shared.c

```diff
@@ -4165,9 +4169,25 @@ struct option long_opts[] = {
 	{"pin-core", required_argument, NULL, LONG_OPT_PIN_CORE},
 	{"timeout", required_argument, NULL, LONG_OPT_TIMEOUT},
 	{"debug-assert", no_argument, &debug_assert, LONG_OPT_DEBUG_ASSERT},
+	{"data-progress-model", required_argument, NULL, LONG_OPT_DATA_PROGRESS},
+	{"control-progress-model", required_argument, NULL, LONG_OPT_CONTROL_PROGRESS},
 	{NULL, 0, NULL, 0},
 };
 
+int ft_parse_progress_model_string(char* progress_str)
+{
+	int ret = -1;
+
+	if (!strncasecmp("FI_PROGRESS_MANUAL", progress_str, 19))
+		ret = FI_PROGRESS_MANUAL;
+	else if (!strncasecmp("FI_PROGRESS_AUTO", progress_str, 17))
+		ret = FI_PROGRESS_AUTO;
+	else if (!strncasecmp("FI_PROGRESS_CONTROL_UNIFIED", progress_str, 28))
+		ret = FI_PROGRESS_CONTROL_UNIFIED;
```

Comment:
Are you including the NULL terminator in these? Is that correct/more correct? Asking because the rest of fabtests just does the length of the characters. Wondering if all of those are wrong

Reply:
I am not including the NULL terminator.  Since I give the max length to compare (specified length, or NULL terminator of progress_str, whichever comes first), I don't think I need to include the NULL terminator.  I am also not doing anything with the progress_str after I parse it besides setting ret = ENUM, if the entire specified string matches.  In this case, I think we are fine.

Reply:
also we are using [getopt_long](https://linux.die.net/man/3/getopt_long) which should always give us null terminated strings in optarg

Reply:
@a-szegel Sorry I should have clarified my comment more - in this code you are adding an extra char to compare, implying you should compare the NULL terminator but I agree, I don't think you need it - ie FI_PROGRESS_MANUAL is 18 characters but you're comparing 19, FI_PROGRESS_AUTO is 16 but you're comparing 17, etc

Reply:
It's usually better to include the null terminator to match the string exactly.  Consider if the user passed in "FI_PROGRESS_MANUAL-v" and left the space out before the -v argument.  Without the null terminator check, this string passes and the -v gets lost without the user knowing that their mistake was ignored.

Reply:
Any reason not to use `strcasecmp()` here? That should behave the same as `strncasecmp()` with the null included.

Reply:
I was more just asking about which convention we should adopt throughout fabtests. Every other use of it in fabtests uses the strncasecmp without the NULL character. I like Jianxin's suggestion of switching all of them to use strcasecmp which seems to handle it explicitly. @a-szegel Can you change this to strcasecmp? I can open a PR to standardize the rest of fabtests.

Reply:
I was just doing `echo FI_PROGRESS_CONTROL_UNIFIED | wc` to count for me, and I guess that got the extra character

Reply:
fixed.